gamemode/modules/base/sv_purchasing.lua

```diff
@@ -423,13 +423,13 @@ local function SetPrice(ply, args)
 
 	local tr = util.TraceLine(trace)
 
-	if not IsValid(tr.Entity) then DarkRP.notify(ply, 1, 4, DarkRP.getPhrase("must_be_looking_at", "gunlab / druglab / microwave")) return "" end
+	if not IsValid(tr.Entity) then DarkRP.notify(ply, 1, 4, DarkRP.getPhrase("must_be_looking_at", "any lab")) return "" end
 
 	local class = tr.Entity:GetClass()
-	if IsValid(tr.Entity) and (class == "gunlab" or class == "microwave" or class == "drug_lab") and tr.Entity.SID == ply.SID then
+	if IsValid(tr.Entity) and (class == "gunlab" or class == "microwave" or class == "drug_lab" or tr.Entity.CanPrice) and tr.Entity.SID == ply.SID then
```

Comment:
CanPrice makes the class checks redundant. 

Also, the name should probably not be "CanPrice". "CanPrice" is not descriptive of what it does.